### ISSUE TITLE
nix-* not in PATH after install

### ISSUE THREAD

**Issue by cscherrer** [NixOS/nix#3317]
Hi,

I installed Nix like this:

```
sh <(curl https://nixos.org/nix/install) --daemon
```

At the end it says,
```
Try it! Open a new terminal, and type:

  $ nix-shell -p nix-info --run "nix-info -m"
```

So I try that (yes, in a new terminal):

```
~ >>> nix-shell -p nix-info --run "nix-info -m"                                
zsh: command not found: nix-shell
```

Here's my `$PATH`:

```
~ >>> echo $PATH                                                          [127]
/home/chad/.zplugin/polaris/bin:/home/chad/.local/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
```

(I'm apparently doing Perl stuff? Weird.)

Any ideas?

I'm running Manjaro Linux 5.4.6 on a Dell XPS 9560. Shell is zsh.

**Comment by matthewbauer** [comment]
Maybe the profile is not getting sourced? Does `source $HOME/.profile` work?

**Comment by cscherrer** [comment]
Thanks @matthewbauer , yes that was it. I don't really understand the role of `.profile`. I don't think zsh ever touches it, seems an odd place to put things for general use. But maybe I'm misunderstanding

**Comment by matthewbauer** [comment]
Yeah this does look like a bug. At least zprofile should be installed by install-nix-from-closure.sh here:

https://github.com/NixOS/nix/blob/8b09105db3869284ee7892f82155dda79f98d6e6/scripts/install-nix-from-closure.sh#L147

According to this, only .zprofile is read at login time:

https://shreevatsa.wordpress.com/2008/03/30/zshbash-startup-files-loading-order-bashrc-zshrc-etc/

/cc @grahamc

**Comment by ngiger** [comment]
I tried the installer on my debian buster using the fish shell. The nix daemon started okay, but after logging out and in, the nix-shell is not found. I only found it via
`sudo find / -name nix-shell
/nix/store/4vz8sh9ngx34ivi0bw5hlycxdhvy5hvz-nix-2.3.7/bin/nix-shell

I fixed this problem by creating `~/.config/fish/config.fish` with the line
`set PATH /nix/var/nix/profiles/default/bin /usr/bin /bin /usr/sbin /sbin /usr/local/bin`

**Comment by muniter** [comment]
I have the same problem, Ubuntu 20.04, Installed but the installer did not modified my `.profile` at all.

**Comment by Hugo-Trentesaux** [comment]
Same thing here on Manjaro. Nix does not get added to path (tried `.bashrc`, `.bash_profile`, `.profile` too)

**Comment by mmsatari** [comment]
In my case using Ubuntu and zsh had to set the `PATH` myself.
`export PATH="$PATH:/nix/var/nix/profiles/default/bin"`

and this path didn't exist at all:
`HOME/.nix-profile/etc/profile.d/nix.sh`

**Comment by jgomo3** [comment]
I'm using Ubuntu.

If I use gnome-terminal, the binaries are not reachable.
But if I use a terminal (Ctrl-Alt-F<1..6>) they work.

The reason, it only work with login shells. Gnome-terminal is a non-login shell (which is correct, it doesn't ask for a password). As a workaround, I can configure gnome-terminal as a login-shell: not ideal.

**Comment by johnmarkwayve** [comment]
I have this problem on Ubuntu 18.04 with `bash`. Not a great start when using one of the most popular distros and the most popular shell doesn't work.

**Comment by volisoft** [comment]
Same with Ubuntu 20.04, bash shell. After exiting terminal nix-* commands are not available.

**Comment by arkanoid87** [comment]
https://nixos.org/guides/install-nix.html and https://nixos.org/download.html shows two different bash commands, the first one is broken on my ubuntu 20.04 installation.
Doing the second on top of the first one fixed the problem.

**Comment by nielsegberts** [comment]
Same on Ubuntu 21.04. Used https://nixos.org/guides/install-nix.html and nix-env was not found. https://nixos.org/download.html did work but requires `. /home/niels/.nix-profile/etc/profile.d/nix.sh` to be run first in every new terminal.

**Comment by Im-Mr-Chris** [comment]
> Maybe the profile is not getting sourced? Does `source $HOME/.profile` work?

This helped me. Thank you.

**Comment by lionello** [comment]
I just installed Nix 2.4 on a brand new MacBook Pro. Nix works in a new terminal tab right after install, but stopped working after a reboot. (But Apple also installed some supplementary OS updates after the 1st reboot so maybe they erased something?) File `/etc/zshrc.backup-before-nix` exists but is identical to `/etc/zshrc` so looks like Nix' changes got undone.

Fixed by creating a new file `~/.zshrc` with the following content:
```sh
fpath=($HOME/.nix-profile/share/zsh/site-functions $fpath)

# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix
```

**Comment by soulomoon** [comment]
Same here look, have no idea where does it get undone

**Comment by soulomoon** [comment]
> I just installed Nix 2.4 on a brand new MacBook Pro. Nix works in a new terminal tab right after install, but stopped working after a reboot. (But Apple also installed some supplementary OS updates after the 1st reboot so maybe they erased something?) File `/etc/zshrc.backup-before-nix` exists but is identical to `/etc/zshrc` so looks like Nix' changes got undone.
> 
> Fixed by creating a new file `~/.zshrc` with the following content:
> 
> ```shell
> fpath=($HOME/.nix-profile/share/zsh/site-functions $fpath)
> 
> # Nix
> if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
>   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
> fi
> # End Nix
> ```

Same here look, have no idea where does it get undone

**Comment by lionello** [comment]
This issue is similar to #1577 (and could be considered a duplicate)

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
